src/pint/simulation.py

```diff
@@ -474,36 +435,175 @@ def calculate_random_models(
             mean_vector = mean_vector[index]
         param_names = cov_matrix.get_label_names(axis=0)
 
-    f_rand = deepcopy(fitter)
-
     # scale by fac
     mean_vector = mean_vector * fac
     scaled_cov_matrix = ((cov_matrix.matrix * fac).T * fac).T
     random_models = []
-    for imodel in range(Nmodels):
+    for _ in range(Nmodels):
+        f_rand = deepcopy(fitter)
```

Comment:
Is there any reason to copy the Fitter and not just the model?

Reply:
Great idea!  And that sped up random model creation a bunch since the potentially large TOA objects aren't being copied, either.  Thanks for that.  I've pushed that up.